### PR TITLE
Support property generation for in arguments.

### LINKED ISSUE #22385: Generate Property CodeAction should know about ref-readonly-ness
After `in` is implemented at call-site, the IDE will be able to know if it was in a ref-readonly context or not, therefore, it can generate a property with that `RefKind`.
This means that `GenerateVariableCodeAction` should accept a `RefKind` in the constructor, and not just `bool _returnsByRef`. Also, the following test should pass if added to `GenerateVariableTests.cs`:

```csharp
        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateVariable)]
        public async Task TestGeneratePropertyInRefReadOnly()
        {
            await TestInRegularAndScriptAsync(
@"
using System;
class Class
{
    void Method(ref readonly int i)
    {
        Method(in this.[|goo|]);
    }
}",
@"
using System;
class Class
{
    public ref readonly int goo => throw new NotImplementedException();

    void Method(ref readonly int i)
    {
        Method(in this.goo);
    }
}", index: 2);
        }
```

cc @VSadov

### PR DESCRIPTION
**Pull request by OmarTawfik**
Closes #22385
Looks like the used API is being renamed in #22472

@dotnet/roslyn-ide @jasonmalinowski for review.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/PublicAPI.Shipped.txt [hunk 1]
```diff
@@ -301,11 +301,8 @@ Microsoft.CodeAnalysis.CSharp.Syntax.ArgumentListSyntax.WithOpenParenToken(Micro
 Microsoft.CodeAnalysis.CSharp.Syntax.ArgumentSyntax
 Microsoft.CodeAnalysis.CSharp.Syntax.ArgumentSyntax.Expression.get -> Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax
 Microsoft.CodeAnalysis.CSharp.Syntax.ArgumentSyntax.NameColon.get -> Microsoft.CodeAnalysis.CSharp.Syntax.NameColonSyntax
-Microsoft.CodeAnalysis.CSharp.Syntax.ArgumentSyntax.RefOrOutKeyword.get -> Microsoft.CodeAnalysis.SyntaxToken
```
Comment (jasonmalinowski) [inline src/Compilers/CSharp/Portable/PublicAPI.Shipped.txt]: This is a breaking change to remove this from shipped. The PublicAPI analyzer has a **REMOVED** syntax you can use in PublicAPI.Unshipped.txt to express this.

**Comment by OmarTawfik** [comment]
More on API change:
For the parse tree, when parsing arguments, you could prefix arguments with `ref` or `out`. The parsed keyword would be stored on `ArgumentSyntax.RefOrOutKeyword`. After introducing readonly references feature, users can now specify `in` on arguments as well.
We need to make an API call here. The current behavior needs to change as this API returns `InKeyword` where applicable today, despite the wrong name.
The PR proposes renaming the public property to `ArgumentSyntax.RefKindKeyword`, and it would behave exactly the same as the old one is behaving today.
Thoughts?

**Comment by jasonmalinowski** [comment]
This as presented would be a breaking change for analyzers, as demonstrated by this PR. I assume that's not being done here without a mitigation of some kind?

**Comment by OmarTawfik** [comment]
@jasonmalinowski  yes. We need to make API call.

**Comment by OmarTawfik** [comment]
cc @dotnet/roslyn-infrastructure [intermittent test failure](https://ci.dot.net/job/dotnet_roslyn/job/post-dev15.5-contrib/job/windows_debug_unit64_prtest/37/testReport/junit/Microsoft.CodeAnalysis.CompilerServer.UnitTests/CompilerServerUnitTests/CscFallBackOutputNoUtf8/)

**Comment by OmarTawfik** [comment]
@dotnet/roslyn-ide for review.

**Comment by OmarTawfik** [comment]
@jasonmalinowski comments addressed. Can you please take a look?

File: src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs [hunk 2]
```diff
@@ -321,6 +321,10 @@ public static bool IsInRefContext(this ExpressionSyntax expression)
             => expression.IsParentKind(SyntaxKind.RefExpression) ||
                (expression?.Parent as ArgumentSyntax)?.RefOrOutKeyword.Kind() == SyntaxKind.RefKeyword;
 
+        public static bool IsInInContext(this ExpressionSyntax expression)
+            => expression.IsParentKind(SyntaxKind.RefExpression) ||
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs]: this does not seem to match the name of the exension.  The extension should probably be called "IsInRefExpressionOrInInContext.

Comment (OmarTawfik) [reply src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs]: Will refactor.

**Comment by OmarTawfik** [comment]
@CyrusNajmabadi comments addressed.

**Comment by OmarTawfik** [comment]
@CyrusNajmabadi @jasonmalinowski any remaining comments I should address here?

File: src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs [hunk 3]
```diff
@@ -321,6 +321,9 @@ public static bool IsInRefContext(this ExpressionSyntax expression)
             => expression.IsParentKind(SyntaxKind.RefExpression) ||
                (expression?.Parent as ArgumentSyntax)?.RefOrOutKeyword.Kind() == SyntaxKind.RefKeyword;
 
+        public static bool IsInInContext(this ExpressionSyntax expression)
+            =>(expression?.Parent as ArgumentSyntax)?.RefKindKeyword.Kind() == SyntaxKind.InKeyword;
```
Comment (jasonmalinowski) [inline src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs]: Space before the (.

File: src/Workspaces/CSharp/Portable/Extensions/ArgumentSyntaxExtensions.cs [hunk 4]
```diff
@@ -15,22 +15,27 @@ internal static class ArgumentSyntaxExtensions
...
             return default;
         }
 
         public static RefKind GetRefKind(this ArgumentSyntax argument)
         {
-            var refSyntaxKind = argument?.RefOrOutKeyword.Kind();
-            return
-                refSyntaxKind == SyntaxKind.RefKeyword ? RefKind.Ref :
-                refSyntaxKind == SyntaxKind.OutKeyword ? RefKind.Out : RefKind.None;
+            switch(argument?.RefKindKeyword.Kind())
```
Comment (jasonmalinowski) [inline src/Workspaces/CSharp/Portable/Extensions/ArgumentSyntaxExtensions.cs]: space after `switch`.

**Pull request merged**